### PR TITLE
doc: bootloader: updated readme structure

### PR DESCRIPTION
**Pull request by ChristopherBittner**
Updated general bootloader page structure.
Updates several procedures and provided more detailed information.

Signed-off-by: Chris Bittner <chris.bittner@nordicsemi.no>

### CODE REVIEW

**Review by hakonfam — APPROVED** [review]

File: doc/nrf/ug_bootloader.rst [hunk 1]
```diff
@@ -47,19 +47,22 @@ The following image shows an abstract representation of the memory layout, assum
 
 For detailed information about the memory layout, see the partition configuration in the :file:`<build folder>/partitions.yml` file or run ``ninja partition_manager_report``.
 
+.. note::
+   To see above information, multi image need to be used. See :ref:`partition_manager` and :ref:`ug_multi_image` for details.
```
Comment (hakonfam) [inline doc/nrf/ug_bootloader.rst:51]:
```suggestion
   The Partition Manager must be enabled for the ``partition_manager_report`` to be available. See :ref:`partition_manager` and :ref:`ug_multi_image` for details.
```

**Review by hakonfam — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_bootloader.rst (hunk 1 above)
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:51]: Would you accept below instead?
The Partition Manager must be enabled for the ``partitions.yml`` and ``partition_manager_report`` to be available. See :ref:`partition_manager` and :ref:`ug_multi_image` for details.

Comment (hakonfam) [reply doc/nrf/ug_bootloader.rst]: I think its important to highlight that one is a file and one is a ninja target.

```
:file:`partitions.yml` and ``partition_manager_report`` target
```

Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst]: I agree, fixed.

File: samples/bootloader/README.rst [hunk 2]
```diff
@@ -127,7 +136,7 @@ Complete the following steps to add the bootloader sample as child image to your
 #. Run ``menuconfig`` on your application to enable Secure Boot:
 
    a. Select :guilabel:`Project` -> :guilabel:`Configure nRF Connect SDK project`.
-   #. Go to :guilabel:`Modules` -> :guilabel:`Nordic nRF Connect` and select :guilabel:`Use Secure Bootloader` to enable :option:`CONFIG_SECURE_BOOT`.
+   #. Go to :guilabel:`Modules` -> :guilabel:`Nordic nRF Connect` -> :guilabel:`Bootloader` and set :guilabel:`Use Secure Bootloader` to enable :option:`CONFIG_SECURE_BOOT`.
```
Comment (mia-ko) [inline samples/bootloader/README.rst:139]: Why the change from select to set..?
I realize that I havent't seen how this actually looks like (a screenshot could probably help). Then it's more difficult to give an advice.

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: samples/bootloader/README.rst (hunk 2 above)
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:139]: It is a checkbox. So you either set or unset it.
I think that select is better with radio buttons.

File: samples/bootloader/README.rst [hunk 3]
```diff
@@ -149,7 +158,8 @@ Complete the following steps to add the bootloader sample as child image to your
 Testing
 =======
 
-To test the bootloader sample, add it to any other sample and build and program that sample it as described above.
+To test the bootloader sample, add it to any other sample, build and program that sample as described in :ref:`bootloader_build_and_run`.
+
 Then test it by performing the following steps:
 
 #. |connect_terminal|
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:165]: Can you elaborate?

File: samples/bootloader/README.rst (hunk 2 above)
Comment (mia-ko) [reply samples/bootloader/README.rst:139]: Then I totally agree with you. Set, check or tick should be used rather than select for checkboxes :)

File: samples/bootloader/README.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bootloader/README.rst:165]: No, just that a numbered list should start with 1. and be continued with #.

File: doc/nrf/ug_bootloader.rst [hunk 4]
```diff
@@ -47,19 +47,22 @@ The following image shows an abstract representation of the memory layout, assum
 
 For detailed information about the memory layout, see the partition configuration in the :file:`<build folder>/partitions.yml` file or run ``ninja partition_manager_report``.
 
+.. note::
+   The Partition Manager must be enabled for the :file:`parititions.yml` and ``partition_manager_report`` target to be available. See :ref:`partition_manager` and :ref:`ug_multi_image` for details.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:51]: What we want to say is that the partitions.yml will not exist and partition_manager_report will not produce any result if Partition Manager is not enabled. In other words they are not targets of Partition Manager but its products. 
Would it still reflect that after your change?

I agree about newline for See(...)

File: doc/nrf/ug_bootloader.rst [hunk 5]
```diff
@@ -47,19 +47,22 @@ The following image shows an abstract representation of the memory layout, assum
...
 
 Immutable bootloader
 ====================
 
 The first step in the chain of trust is an immutable first stage bootloader, in the following referred to as immutable bootloader.
-The code abbreviates this bootloader as SB (for Secure Boot) or B0.
+The code abbreviates this bootloader as SB (Secure Boot, Secure Bootloader) or B0.
 
 The immutable bootloader runs after every reset and establishes the root of trust by verifying the signature and metadata of the next image in the boot sequence.
 If verification fails, the boot process stops.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:62]: Fixed.

File: doc/nrf/ug_bootloader.rst [hunk 6]
```diff
@@ -47,19 +47,22 @@ The following image shows an abstract representation of the memory layout, assum
...
 
-This way, the immutable bootloader can guarantee that if the flash of the next image in the boot sequence (another bootloader or the application) has been tampered with in any way, that image will not start up.
-So if an attacker attempts to take over the device by altering the flash, the device will not boot and not run the infected code.
+This way, the immutable bootloader can guarantee that if the flash of the next image in the boot sequence (another bootloader or an application) has been tampered with in any way, that image will not start up.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:64]: I think _that_ in the context here was to point to a specific image i.e. _that image_.

File: doc/nrf/ug_bootloader.rst [hunk 7]
```diff
@@ -84,9 +87,9 @@ The upgradable bootloader should carry out the same signature and metadata verif
 In addition, it can provide functionality for upgrading both itself and the following image in the boot sequence (in most cases, the application).
 
 There are two partitions where the upgradable bootloader can be stored: slot 0 and slot 1 (also called *S0* and *S1*).
-A new bootloader image is stored in the partition that is not currently used.
+A new bootloader image is placed in the partition that is not currently used.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:90]: Fixed.

File: doc/nrf/ug_bootloader.rst [hunk 8]
```diff
@@ -84,9 +87,9 @@ The upgradable bootloader should carry out the same signature and metadata verif
...
 When booting, the immutable bootloader checks the version information for the images in slot 0 and slot 1 and boots the one with the highest version.
-If this image is faulty and cannot be booted, the other partition will always hold a working image, and this one is booted instead.
+If this image is faulty and cannot be booted, the other partition will always hold a working image, and that one is booted instead.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:92]: Fixed.

File: doc/nrf/ug_bootloader.rst [hunk 9]
```diff
@@ -96,7 +99,7 @@ These signed variants can be used to perform an upgrade procedure through the :r
 Adding a bootloader chain to your application
 *********************************************
 
-The |NCS| includes a sample implementation of an immutable bootloader.
+The |NCS| includes a sample implementation of an :ref:`bootloader`.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:102]: Fixed.

File: doc/nrf/ug_bootloader.rst [hunk 10]
```diff
@@ -96,7 +99,7 @@ These signed variants can be used to perform an upgrade procedure through the :r
...
 Additionally, the |NCS| comes with a slightly modified version of :doc:`mcuboot:index`.
 
 Both bootloaders can easily be included in your application using :ref:`ug_multi_image`.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:105]: Fixed.

File: doc/nrf/ug_bootloader.rst [hunk 11]
```diff
@@ -47,19 +47,22 @@ The following image shows an abstract representation of the memory layout, assum
...
-The code abbreviates this bootloader as SB (for Secure Boot) or B0.
+The code abbreviates this bootloader as SB (Secure Boot, Secure Bootloader) or B0.
 
 The immutable bootloader runs after every reset and establishes the root of trust by verifying the signature and metadata of the next image in the boot sequence.
 If verification fails, the boot process stops.
 
-This way, the immutable bootloader can guarantee that if the flash of the next image in the boot sequence (another bootloader or the application) has been tampered with in any way, that image will not start up.
-So if an attacker attempts to take over the device by altering the flash, the device will not boot and not run the infected code.
+This way, the immutable bootloader can guarantee that if the flash of the next image in the boot sequence (another bootloader or an application) has been tampered with in any way, that image will not start up.
+For example, if an attacker attempts to take over the device by altering the flash, the device will not boot and not run the infected code.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:65]: Fixed.

File: samples/bootloader/README.rst [hunk 12]
```diff
@@ -58,10 +66,10 @@ The flash layout is defined by :file:`samples/bootloader/pm.yml`.
...
 The bootloader sample defines four main areas:
 
-1. **B0** - Contains the bootloader sample.
-#. **Provision** - Stores the provisioned data.
-#. **S0** - One of two potential storage areas for the second stage bootloader.
-#. **S1** - One of two potential storage areas for the second stage bootloader.
+1. **B0** - contains the bootloader sample.
+#. **Provision** - stores the provisioned data.
+#. **S0** - one of two potential storage areas for the second stage bootloader.
+#. **S1** - one of two potential storage areas for the second stage bootloader.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:72]: Ok, good to know. So I have reverted back to capital letters and removed periods.

File: samples/bootloader/README.rst [hunk 13]
```diff
@@ -70,11 +78,12 @@ Provisioning
 ============
 
 The public key hashes are not compiled in with the source code of the bootloader sample.
-Instead, they must be stored in a separate memory region through a process called *provisioning*.
+Instead, they must be stored in  separate memory region through a process called *provisioning*.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:81]: Good catch. That wasn't intended. 
Fixed.

File: samples/bootloader/README.rst [hunk 14]
```diff
@@ -70,11 +78,12 @@ Provisioning
...
 
 By default, the bootloader sample will automatically generate and provision public key hashes directly into the bootloader HEX file, based on the specified private key and additional public keys.
+
 Alternatively, to facilitate the manufacturing process of a device with the bootloader sample, it is possible to decouple this process and program the sample HEX file and the HEX file containing the public key hashes separately.
-If you choose to do so, use the Python scripts in ``scripts\bootloader`` to create and provision the keys manually.
+If you choose to do so, use the Python scripts located in ``scripts\bootloader`` to create and provision the keys manually.
 
    .. note::
       On some chips (for example, nRF9160 or nRF5340), the provisioned data is held in the OTP region in UICR.
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:89]: Yes, I wanted to do the same here but then I assumed the author' intention was not to join them. Probably to have the option to independently remove them from the documentation when countermeasures will be implemented. I will leave it as it is for now :)

File: samples/bootloader/README.rst [hunk 15]
```diff
@@ -117,7 +126,7 @@ Complete the following steps to add the bootloader sample as child image to your
        openssl ecparam -name prime256v1 -genkey -noout -out priv.pem
```
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:126]: Fixed.

File: samples/bootloader/README.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply samples/bootloader/README.rst:165]: Haha, yes.
Fixed.

File: doc/nrf/ug_bootloader.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/nrf/ug_bootloader.rst]: Oooh, no, that changes it completely! I understood it the other way around.
In that case, how about:

The Partition Manager must be enabled to make the :file:`parititions.yml` file and the ``partition_manager_report`` target available.
See :ref:`partition_manager` and :ref:`ug_multi_image` for details.

File: doc/nrf/ug_bootloader.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/nrf/ug_bootloader.rst:65]: Ah, I see :)
Still it "sounded" like something was missing... (probably just me) Would it be better if simply said:
"...in any way, the image will not..."?

File: samples/bootloader/README.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/bootloader/README.rst:89]: No problem to later remove this, combined or not :) But I see your point. Leave as is for now, ok for me!

File: doc/nrf/ug_bootloader.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst:65]: Agree, fixed.

File: doc/nrf/ug_bootloader.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (ChristopherBittner) [reply doc/nrf/ug_bootloader.rst]: Fixed.

**Review by hakonfam — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]
